src/tests/unittests/asn1.c

```diff
@@ -47,12 +47,91 @@ static void torture_large_oid(void **state)
 	assert_int_equal(rv, SC_ERROR_NOT_SUPPORTED);
 }
 
+static void torture_integer(void **state)
+{
+	/* Without the Tag and Length {0x02, 0x01} */
+	u8 zero[] = {0x00};
+	u8 one[] = {0x01};
+	u8 minus_one[] = {0xFF};
+	u8 max2[] = {0x7F, 0xFF};
+	u8 min2[] = {0x80, 0x00};
+	u8 max4[] = {0x7F, 0xFF, 0xFF, 0xFF};
+	u8 min4[] = {0x80, 0x00, 0x00, 0x00};
+	u8 over[] = {0x7F, 0xFF, 0xFF, 0xFF, 0xFF};
+	int value;
+	int rv = 0;
+
+	rv = sc_asn1_decode_integer(zero, sizeof(zero), &value);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(value, 0);
+
+	rv = sc_asn1_decode_integer(one, sizeof(one), &value);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(value, 1);
```

Comment:
again, separate scenarios, so they should be separate test cases

Reply:
Here, you mean encode/decode as separate scenarios or different values/errors? If the first, I wanted to avoid having the "magic numbers" all over the place copied.

Reply:
I mean that
```

	rv = sc_asn1_decode_integer(zero, sizeof(zero), &value);
	assert_int_equal(rv, SC_SUCCESS);
	assert_int_equal(value, 0);
	rv = asn1_encode_integer(value, &buf, &buflen);
	assert_int_equal(rv, SC_SUCCESS);
	assert_int_equal(buflen, sizeof(zero));
	assert_memory_equal(buf, zero, buflen);
	free(buf);
```
is a single test case
then 
```

	rv = sc_asn1_decode_integer(one, sizeof(one), &value);
	assert_int_equal(rv, SC_SUCCESS);
	assert_int_equal(value, 1);

	rv = asn1_encode_integer(value, &buf, &buflen);
	assert_int_equal(rv, SC_SUCCESS);
	assert_int_equal(buflen, sizeof(one));
	assert_memory_equal(buf, one, buflen);
	free(buf);
```
is a second test case, and so on...